scanner.go

```diff
@@ -201,6 +202,25 @@ func (s *Scanner) consumeUntil(b byte) (string, bool) {
 	return "", false
 }
 
+// parseOctalString parses the digits of an octal escape sequence, returning the
+// run it escapes.
+//
+// depth is the number of chars that we have parsed so far.
+// out is the current value of the integer that we are building.
+// We only call this function the first time when we know that we are looking at
+// at least one octal digit following a backslash '\'.
+func (s *Scanner) parseOctalString(depth, out int) (rune, error) {
+	if depth == 3 || s.isEOF(0) {
```

Comment:
Could this be a for loop instead of being recursive? One option is to build a version of `consumeUntil` that takes a closure argument so you can slurp up at most three octal integers? I would also probably just inline the loop into the switch below, for consistency. `strconv.ParseInt` can handle octal, so I'd use that instead of carefully building up the integer ourselves, given we can actually avoid it.

Reply:
Okay, it's now an inlined for-loop instead of a recursive function and it calls stringconv.ParseUint() to do the heavy lifting. The code is sufficiently simple that adding a closure probably would have only obfuscated things.

Reply:
Thanks. :)